runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -234,7 +228,11 @@ public void write(IToken token, Object data) throws OverwriteException {
         if (!runtime.getParameters().isCacheDisabled()) {
             ILogData data;
             try {
-                data = readCache.get(address);
+                data = readCache.getIfPresent(address);
+                if (data == null) {
+                    final ILogData loadedVal = fetch(address);
+                    data = readCache.get(address, () -> loadedVal);
```

Comment:
For future reference can you please add a comment why we check the read cache again upon return, and how it relates to NRE.